tests/runner.py

```diff
@@ -956,15 +952,19 @@ def do_run(self, src, expected_output, args=[], output_nicerizer=None,
       basename = 'src.c'
       Building.COMPILER = to_cc(Building.COMPILER)
 
-    dirname = self.get_dir()
-    filename = os.path.join(dirname, basename)
-    if not no_build:
-      self.build(src, dirname, filename, main_file=main_file, additional_files=additional_files, libraries=libraries, includes=includes,
+    if no_build:
+      js_file = src
```

Comment:
this changes the current behavior, doesn't it? "no_build" means "we already built, and the output is in filename.o.js, as if we did build" but this now implies we don't need the suffix.

the idea sounds good, but I'm not sure this catches all the callers that might be affected by that difference - perhaps explains the test failures?

Reply:
Ah yes, there are a whole bunch of tests failing for this reason.

I think I can clean this up.